### ISSUE TITLE
Issue with InternalTransitionIf and trigger with single parameter

### ISSUE THREAD

**Issue by matt-w-cotton** [dotnet-state-machine/stateless#272]
First off, thank you for a terrific library.
I have yet to track down the exact cause of the issue, but I assume it popped up when adding the `Func<object[], bool>` InternalTransitionIf guards.

Basically now when you attempt to `GetPermittedTriggers()`, if you have InternalTransitionIf, using trigger with single param, you will get:
`System.ArgumentException : An argument of type System.String is required in position 0.`

I added a quick test to demonstrate the issue.  Same test with 2 params and the issue does not come up.  `Unpack` is never called and thus never throws.
Here is the example:
```[Fact]
public void ConditionalInternalTransitionWithOneParam_ShouldWork()
{
    var isPermitted = true;
    var sm = new StateMachine<State, Trigger>(State.A);

    var trig = sm.SetTriggerParameters<string>(Trigger.X);
    sm.Configure(State.A)
        .InternalTransitionIf(trig, u => isPermitted, (u, t) => { });

    Assert.Equal(1, sm.GetPermittedTriggers().ToArray().Length);
    isPermitted = false;
    Assert.Equal(0, sm.GetPermittedTriggers().ToArray().Length);
}
```

and stack trace:
```System.ArgumentException : An argument of type System.String is required in position 0.
Stack Trace:
C:\dev\main\stateless\src\Stateless\ParameterConversion.cs(15,0): at Stateless.ParameterConversion.Unpack(Object[] args, Type argType, Int32 index)
C:\dev\main\stateless\src\Stateless\ParameterConversion.cs(26,0): at Stateless.ParameterConversion.Unpack[TArg](Object[] args, Int32 index)
C:\dev\main\stateless\src\Stateless\TransitionGuard.cs(19,0): at Stateless.StateMachine`2.TransitionGuard.<>c__DisplayClass4_0`1.<ToPackedGuard>b__0(Object[] args)
C:\dev\main\stateless\src\Stateless\TransitionGuard.cs(111,0): at Stateless.StateMachine`2.TransitionGuard.<>c__DisplayClass17_0.<UnmetGuardConditions>b__0(GuardCondition c)
at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)          
at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
C:\dev\main\stateless\src\Stateless\TransitionGuard.cs(110,0): at Stateless.StateMachine`2.TransitionGuard.UnmetGuardConditions(Object[] args)
C:\dev\main\stateless\src\Stateless\TriggerBehaviour.cs(49,0): at Stateless.StateMachine`2.TriggerBehaviour.UnmetGuardConditions(Object[] args)
C:\dev\main\stateless\src\Stateless\StateRepresentation.cs(290,0): at Stateless.StateMachine`2.StateRepresentation.<>c__DisplayClass69_0.<GetPermittedTriggers>b__1(TriggerBehaviour a)
at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
C:\dev\main\stateless\src\Stateless\StateRepresentation.cs(290,0): at Stateless.StateMachine`2.StateRepresentation.<>c__DisplayClass69_0.<GetPermittedTriggers>b__0(KeyValuePair`2 t)
at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()          
at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
C:\dev\main\stateless\src\Stateless\StateRepresentation.cs(296,0): at Stateless.StateMachine`2.StateRepresentation.GetPermittedTriggers(Object[] args)
C:\dev\main\stateless\src\Stateless\StateMachine.cs(129,0): at Stateless.StateMachine`2.GetPermittedTriggers(Object[] args)
```

Thanks again.

**Comment by matt-w-cotton** [comment]
Hi @user81lgp, due to time constraints I gave up on trying to fix the stateless code.  For now we just worked around the issue by using the 2 argument version of the `TriggerWithParameters` and toss out the second argument.

**Comment by smartcaveman** [comment]
@user81lgp  @matt-w-cotton @HenningNT - I submitted a PR with a patch and related tests that demonstrate the conditions under which the problem happens.  There are still some issues related to calling `StateMachine.GetPermittedTriggers(object[])`.  The intent of what that method **should do** was not very clear to me, but these changes will fix cases where `StateMachine.ToString()`, `StateMachine.PermittedTriggers` and `StateMachine.CanFire(TTrigger)` all currently throw unexpected exceptions.

PR: https://github.com/dotnet-state-machine/stateless/pull/276

**Comment by HenningNT** [comment]
Will be fixed in 5.1.0.
